lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs

```diff
@@ -598,6 +599,38 @@ spec = describe "NEW_SHELLEY_TRANSACTIONS" $ do
             , expectField #outputs (`shouldNotContain` [expectedTxOutSource])
             ]
 
+    it "TRANS_NEW_CREATE_04ab - Constructed inputs = Decoded inputs" $ \ctx -> runResourceT $ do
+        wa <- fixtureWallet ctx
+        wb <- emptyWallet ctx
+        let amt = (minUTxOValue (_mainEra ctx) :: Natural)
+        payload <- liftIO $ mkTxPayload ctx wb amt
+
+        rTx <- request @(ApiConstructTransaction n) ctx
+            (Link.createUnsignedTransaction @'Shelley wa) Default payload
+        verify rTx [ expectSuccess ]
+
+        let expectedInputs = getFromResponse (#coinSelection . #inputs) rTx
+        let txCbor = getFromResponse #transaction rTx
+
+        let decodePayload = Json (toJSON $ ApiSerialisedTransaction txCbor)
+        rDecodedTxSource <- request @(ApiDecodedTransaction n) ctx
+            (Link.decodeTransaction @'Shelley wa) Default decodePayload
+
+        let decodedInputs = getFromResponse #inputs rDecodedTxSource
+
+        let testEquality
+                [ApiCoinSelectionInput id' index' address' path' amt' assets']
+                [WalletInput (ApiWalletInput id'' index'' address'' path'' amt'' assets'')] =
+                id' == id'' &&
+                index' == index'' &&
+                address' == address'' &&
+                path' == path'' &&
+                amt' == amt'' &&
+                assets' == assets''
+            testEquality _ _ = False
```

Comment:
Perhaps we could replace this `testEquality` function with `==` if we merge the `ApiCoinSelectionInput` and `ApiWalletInput` types. See https://github.com/input-output-hk/cardano-wallet/pull/3012

Reply:
indeed, we can simplify considerably this testEquality